contrib/pkg/installmanager/installmanager.go

```diff
@@ -0,0 +1,242 @@
+package installmanager
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// metadataRelativePath is the location of the installers cluster metadata file
+	// relative to our WorkDir.
+	metadataRelativePath = "metadata.json"
+)
+
+type InstallManager struct {
+	log         log.FieldLogger
+	LogLevel    string
+	WorkDir     string
+	ClusterName string
+	Namespace   string
+	KubeClient  clientset.Interface
+}
+
+// NewInstallManagerCommand is the entrypoint to create the 'install-manager' subcommand
+func NewInstallManagerCommand() *cobra.Command {
+	opts := &InstallManager{}
+	cmd := &cobra.Command{
+		Use:   "install-manager NAMESPACE CLUSTER_NAME",
+		Short: "Executes and oversees the openshift-installer.",
+		Long:  "The Hive Install Manager runs the phases of the openshift-installer, edits generated assets before completing install, and monitors for artifacts that need to be uploaded back to Hive.",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.complete(args); err != nil {
+				log.WithError(err).Error("cannot complete command")
+				return
+			}
+
+			if len(args) != 2 {
+				cmd.Help()
+				opts.log.Fatal("invalid command arguments")
+			}
+			// Parse the namespace/name for our cluster deployment:
+			opts.Namespace, opts.ClusterName = args[0], args[1]
+
+			if err := opts.validate(); err != nil {
+				log.WithError(err).Error("invalid command options")
+				return
+			}
+			if err := opts.Run(); err != nil {
+				log.WithError(err).Error("runtime error")
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&opts.LogLevel, "log-level", "info", "log level, one of: debug, info, warn, error, fatal, panic")
+	flags.StringVar(&opts.WorkDir, "work-dir", "/output", "directory to use for all input and output")
+	return cmd
+}
+
+func (m *InstallManager) complete(args []string) error {
```

Comment:
Make public (that would allow executing this command from another command, if needed)